doc/adrs/098_use_the_editing_context_to_compute_the_metamodels.adoc

```diff
@@ -0,0 +1,29 @@
+= ADR-098 - Use the editing context to compute the metamodels
+
+== Context
+
+Up until now, in Sirius Web, the metamodels available for an editing context were a mix of global metamodels available all the time and the domains converted when the editing context was loaded.
+
+There are no specific features related to the metamodels converted from a domain.
+Only the global metamodels have some dedicated features such as the creation of dedicated resources (robot flow, domain and view sample etc)
+
+If we ever want to consider specific features for any metamodel, we should embrace the fact that, in Sirius Web, all the editing contexts do not have the same metamodels.
+
+== Decision
+
+In Sirius Web, we will only use the `EPackage.Registry` of the editing context in order to find the available metamodels.
+Sirius Components will stop considering a set of global EPackages as special and available everywhere.
+
+In order to force ourselves to properly consider the impact of this use case, some metadata will be added to Sirius Web projects.
+We will only keep a list of natures in these project metadata for now.
+A nature will defined as an identifier.
+
+We will introduce the nature `siriusWeb://nature?kind=studio` in order to identify Sirius Web projects created using the studio template.
+
+Those projects will be the only one with the domain and view metamodels and no other metamodels.
+As a result, it should only be possible to create domains and views in these projects and nothing else.
```

Comment:
After a small chat with @sbegaudeau, it seems we want the contrary. A project with the studio nature can do anything while a project without the studio nature cannot manipulate domain or view models

Reply:
I changed the rule with these new conditions.